File: api/Models/Race.cs
namespace api.Models
{
    public class Race
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public DateTime DateTime { get; set; }
        public Season Season { get; set; }
        public Guid SeasonId { get; set; }
        public List<Result> Results { get; set; }
    }
}

File: car-racing-tournament-web/src/app/models/race.ts
import { Time } from "@angular/common";
import { Result } from "./result";
import { Season } from "./season";

export class Race {
  constructor(
    public id: string,
    public name: string,
    public dateTime: Time,
    public season: Season,
    public results: Result[]
  ) {}
}
